File: API/Entities/Member.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace API.Entities
{
    public class Member
    {
        public string Id { get; set; } = null!;
        public required string DisplayName{ get; set; }
        public string? ImageUrl { get; set; }
         public string? Description { get; set; }
        public required string Gender { get; set; }
        public required string City { get; set; }
        public required string Country { get; set; }
        public DateOnly DateOfBirth { get; set; }
        public DateTime Created { get; set; } = DateTime.UtcNow;
        public DateTime LastActive { get; set; } = DateTime.UtcNow;

        [JsonIgnore]
        public List<Photo> Photos { get; set; } = [];

        [JsonIgnore]
        [ForeignKey(nameof(Id))]
        public AppUser AppUser { get; set; } = null!;
    }
}

File: Client/src/types/member.ts
export type Member = {
  id: string
  displayName: string
  imageUrl?: string
  description?: string
  gender: string
  city: string
  country: string
  dateOfBirth: string
  created: string
  lastActive: string
}

export type Photo = {
  id: number
  url: string
  publicId: string
  member: string
  memberId: string
}

export type EditableMember = {
  displayName : string, 
  city: string, 
  country: string,
  description?: string
}
